### PR TITLE
[android] Remove useless interfaces

### PR DESCRIPTION
**Pull request by rtsisyk**

### CODE REVIEW

File: android/app/src/main/java/app/organicmaps/MwmActivity.java [hunk 1]
```diff
@@ -515,18 +511,6 @@ private int getDownloadMapsCounter()
     return info == null ? 0 : info.filesCount;
   }
 
-  @Override
-  public void onNoConnectionError()
-  {
-    dismissAlertDialog();
-    mAlertDialog = new MaterialAlertDialogBuilder(this, R.style.MwmTheme_AlertDialog)
-        .setTitle(R.string.common_check_internet_connection_dialog_title)
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/MwmActivity.java:523]: Was this functionality broken/lost during refactorings? Why is it removed? Maybe it should be restored instead. What was it about compared to 2020 mapsme version?

File: android/app/src/main/java/app/organicmaps/util/Utils.java [hunk 2]
```diff
@@ -347,13 +347,10 @@ public static void sendFeedback(@NonNull Activity activity)
...
 
-  public static void navigateToParent(@Nullable Activity activity)
+  public static void navigateToParent(@NonNull Activity activity)
   {
-    if (activity == null)
-      return;
-
-    if (activity instanceof CustomNavigateUpListener)
-      ((CustomNavigateUpListener) activity).customOnNavigateUp();
+    if (activity instanceof MwmActivity)
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/util/Utils.java:352]: A comment why this custom logic is needed and when it happens would be helpful.

File: android/app/src/main/java/app/organicmaps/MwmActivity.java (hunk 1 above)
Comment (rtsisyk) [reply android/app/src/main/java/app/organicmaps/MwmActivity.java:523]: The only one caller of `onNoConnectionError()` was AbstractGuidesClickListener:

https://github.com/mapsme/omim/blob/1892903b63f2c85b16ed4966d21fe76aba06b9ba/android/src/com/mapswithme/maps/maplayer/guides/AbstractGuidesClickListener.java#L27-L38

File: android/app/src/main/java/app/organicmaps/util/Utils.java (hunk 2 above)
Comment (rtsisyk) [reply android/app/src/main/java/app/organicmaps/util/Utils.java:353]: It seems the initial concept was for Fragments like EditorFragment, SearchFragment, and others to run within their dedicated activities (EditorActivity, SearchActivity, etc.) in phone mode or as part of MwmActivity in tablet mode. I have doubts that dedicated activities are needed at all, but this is a different topic to research that can handled as part of table UI improvement0. 

As part of changes done by this PR, MwmActivity was the only class that implemented CustomNavigateUpListener. Therefore, I simply removed the interface without changing the navigation logic.

**Review by biodranik — APPROVED** [review]
LGTM if it didn't break something.
CC @Jean-BaptisteC

**Review by Jean-BaptisteC — APPROVED** [review]
Pixel 6 - Android 13
✅ Map Layer
✅ Theme switch
✅ Editor
✅ Bookmark
✅ Copyright fragment

Tell me if you want I test others things

**Comment by rtsisyk** [comment]
> Pixel 6 - Android 13 ✅ Map Layer ✅ Theme switch ✅ Editor ✅ Bookmark ✅ Copyright fragment
> 
> Tell me if you want I test others things

Thanks a lot! More cleanups are coming soon in the next patch. This one was relatively safe.

**Pull request merged**